_playwright-tests/style_guide.md

```diff
@@ -0,0 +1,141 @@
+   - Use steps to describe what the test does in more detail, these show up in the test report.
+   - Don't use comments to describe the test, as they will not be visible in the report (only use them if you need to explain the code further, not the test).
+
+8. **Helpers and fixtures**: \
+   Extend Playwright with custom functionality only when the abstracted thing is generic enough.
+   - Good examples: cleanup of resources before and after a test, navigation to your app with auto retrying, locating a row in a table.
+   - Bad examples: page objects, clicking on a button, filling in a form (that's the core of the test, if you want to use that for setting up the test environment use the API).
+
+9. **API tests**: \
+   If you are heavily interacting with your API, or writing API tests, consider using an openapi generator to generate the API client and a fixture to set up the API client for all tests that need it.
+
+## Test Structure 🏗
+Recommended folder/file structure, this will make it easier to run multiple test suites and once and share helpers/fixtures if desired.
+
+```
+/_playwright-tests
+  /utils             # Testing utilities
+    /helpers         # Helper functions/snippets
+    /fixtures        # Playwright automatic fixtures
+  /UI                # UI/API depending on the repository, containing the tests
+    some-feature.spec.ts
+```
+
+Each test file should follow this general structure:
+
+```typescript name=example.spec.ts
+import { test, expect } from '@playwright/test';
+import { setupTest, closePopups } from '../utils/helpers';
+
+test.describe('Feature Name', () => {
+  test.beforeEach(async ({ page }) => {
+    // Common setup for each test
+    await setupTest(page);
+  });
+
+  test('should perform some action correctly', async (page, customFixture) => {
+    test.step('naviagat to page', async () => {
+      await yourPage.navigate();
+      await closePopups(page);
+    });
+
+    test.step('do something and verify it', async () => {
+      await yourPage.getByRole().action();
+      await expect(yourPage.resultElement).toBeVisible();
+    });
+
+    // More steps...
+  });
+
+  // More tests...
+});
+```
+
+## Selectors 🔍
+
+In order of preference, use these selector strategies:
+
+1. **Roles with name**
+
+   This is highly preferred, because this selector actually reflects how
+   users and assistive technology perceive the page, i.e.: tests user-visible behavior.
+   Beware that the name is an [accessible name](https://w3c.github.io/accname/#dfn-accessible-name), not just the HTML 'name' attribute.
+
+   ```typescript
+   page.getByRole('button', { name: 'Submit' });
+   ```
+
+2. **Text content** (when applicable)
+   If possible, use with the `exact` option set to true.
+
+   ```typescript
+   page.getByText('Welcome to the application');
+   ```
+
+3. **Test IDs** (for hard to locate elements)
+
+   ```typescript
+   page.locator('data-testid=submit-button');
```

Comment:
I like this example, but tests ids have a few ways to get them. 
This locator should be highlighted to target specific element variables.